paymentsheet/src/main/java/com/stripe/android/link/ui/signup/SignUpScreen.kt

```diff
@@ -244,10 +244,15 @@ private fun SecondaryFields(
         PrimaryButton(
             modifier = Modifier.padding(vertical = 16.dp),
             label = stringResource(R.string.stripe_link_sign_up),
-            state = if (signUpScreenState.signUpEnabled) {
-                PrimaryButtonState.Enabled
-            } else {
-                PrimaryButtonState.Disabled
+            state = when {
+                signUpScreenState.submitState == SignUpScreenState.SubmitState.Submitting ->
+                    PrimaryButtonState.Processing
+                signUpScreenState.submitState == SignUpScreenState.SubmitState.Success ->
+                    PrimaryButtonState.Completed
```

Comment:
We should probably just set the button state back to `PrimaryButtonState.Enabled` on success. Otherwise, we get the “payment completed” animation, which feels incorrect.